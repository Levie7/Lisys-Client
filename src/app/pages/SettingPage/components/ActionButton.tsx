import * as React from 'react';

import { Button, ButtonType } from 'src/shared/components/Button';
import { Dropdown } from 'src/shared/components/Dropdown';
import { Icon } from 'src/shared/components/Icon';
import { Menu } from 'src/shared/components/Menu';

interface ActionButtonProps extends React.HTMLAttributes<any> {
    action: 'create' | 'read' | 'update' | 'delete' | 'back' | 'list' | 'action';
    buttonType: ButtonType;
    iconType?: string;
    section: string;
    title: string;

    setAction?: (action: string) => void;
    setSection: ({ action, section }: { action: string; section: string }) => void;
}

export const ActionButton = ({
    action,
    buttonType,
    className,
    iconType,
    section,
    setAction,
    setSection,
    title,
}: ActionButtonProps) => {
    function handleClick() {
        setSection({ action, section });
    }

    function handleMenuClick(e: any) {
        setAction!(e.key);
    }

    function renderActions() {
        let menu = (
            <Menu onClick={handleMenuClick}>
                <Menu.Item key='Active'>Activate</Menu.Item>
                <Menu.Item key='Inactive'>Inactivate</Menu.Item>
            </Menu>
        );

        return (
            <Dropdown overlay={menu}>
                <Button className={className}>Actions {Icon['action']}</Button>
            </Dropdown>
        );
    }

    function renderDefault() {
        return (
            <Button className={className} onClick={handleClick} type={buttonType}>
                {iconType && Icon[iconType]} {title}
            </Button>
        );
    }

    return action === 'action' ? renderActions() : renderDefault();
};
