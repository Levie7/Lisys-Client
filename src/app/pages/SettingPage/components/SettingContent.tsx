import * as React from 'react';

import { CompanyForm } from '../containers/CompanyForm';
import { GeneralForm } from '../containers/GeneralForm';
import { UserManagement } from '../containers/UserManagement';
import { SettingContentHeader } from './SettingContentHeader';

interface SettingContentProps {
    category: string;
    isDefault: boolean;
    title: string;
}

export const SettingContent = React.memo<SettingContentProps>(({ category, isDefault, title }) => {
    function renderDefaultForm(category: string) {
        switch (category) {
            case 'company':
                return <CompanyForm />;
            case 'general':
                return <GeneralForm />;
            default:
                return null;
        }
    }

    return (
        <div id='SettingContent'>
            {isDefault ? (
                <>
                    <SettingContentHeader title={title} />
                    {renderDefaultForm(category)}
                </>
            ) : (
                <UserManagement />
            )}
        </div>
    );
});
