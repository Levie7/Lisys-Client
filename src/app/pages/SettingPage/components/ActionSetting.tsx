import * as React from 'react';

import { Button } from 'src/shared/components/Button';

export class ActionSetting extends React.Component<{}> {
    render() {
        return (
            <div className='d-flex fj-end'>
                <Button type='primary'>Save</Button>
            </div>
        );
    }
}
