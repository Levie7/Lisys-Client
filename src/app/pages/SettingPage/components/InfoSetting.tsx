import * as React from 'react';

import { Divider } from 'src/shared/components/Divider';

interface InfoSettingProps {
    children?: React.ReactNode;
    description?: string;
    title?: string;
}

export const InfoSetting = React.memo<InfoSettingProps>(({ children, description, title }) => (
    <>
        <Divider />
        <div className='d-flex fd-column fd@md-row'>
            <div className='col-12 col@md-4 mb-4'>
                <h1 className='fw-bold'>{title}</h1>
                {description}
            </div>
            <div className='col-12 col@md-4'>{children}</div>
        </div>
    </>
));
