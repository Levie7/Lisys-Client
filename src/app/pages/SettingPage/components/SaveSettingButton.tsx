import * as React from 'react';

import { Button } from 'src/shared/components/Button';
import { UIContext } from 'src/shared/contexts/UIContext';

export const SaveSettingButton = React.memo(() => (
    <UIContext.Consumer>
        {({ isMobile }) => (
            <Button className={isMobile ? 'w-100' : ''} htmlType='submit' type='primary'>
                Save
            </Button>
        )}
    </UIContext.Consumer>
));
