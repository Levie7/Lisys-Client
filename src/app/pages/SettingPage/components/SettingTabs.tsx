import * as React from 'react';

import { TabPositions, Tabs } from 'src/shared/components/Tabs';

import { tabData } from '../constants';
import { SettingContent } from './SettingContent';
import { SettingTabPaneItem } from './SettingTabPaneItem';

interface SettingTabsProps {
    position: TabPositions;
}

function SettingTabsPure({ position }: SettingTabsProps) {
    return (
        <Tabs defaultActiveKey='3' tabPosition={position}>
            {tabData.map((tab) => (
                <Tabs.TabPane
                    key={tab.key}
                    tab={
                        <SettingTabPaneItem
                            description={tab.description}
                            icon={tab.icon}
                            title={tab.title}
                        />
                    }
                >
                    <SettingContent
                        category={tab.category}
                        isDefault={tab.type === 'default'}
                        title={tab.title}
                    />
                </Tabs.TabPane>
            ))}
        </Tabs>
    );
}

export const SettingTabs = React.memo(SettingTabsPure);
