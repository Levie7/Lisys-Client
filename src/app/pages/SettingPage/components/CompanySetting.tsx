import * as React from 'react';

import { Divider } from 'src/shared/components/Divider';
import { Form } from 'src/shared/components/Form';
import { Input } from 'src/shared/components/Input';
import { Upload } from 'src/shared/components/Upload';

import { formItemLayout } from '../constants';
import { ActionSetting } from './ActionSetting';

export class CompanySetting extends React.Component<{}> {
    render() {
        return (
            <>
                <Divider orientation='left'>Company Information</Divider>
                <Form {...formItemLayout}>
                    <Form.Item label='Company Name'>
                        <Input placeholder='PT. Lisys' id='name' />
                    </Form.Item>
                    <Form.Item label='Company Year'>
                        <Input placeholder='2019' id='year' />
                    </Form.Item>
                    <Form.Item label='Company Logo'>
                        <Upload name='company_logo' listType='picture-card' />
                    </Form.Item>
                    <Divider />
                    <ActionSetting />
                </Form>
            </>
        );
    }
}
