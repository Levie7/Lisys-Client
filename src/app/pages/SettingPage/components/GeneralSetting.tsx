import * as React from 'react';

import { Divider } from 'src/shared/components/Divider';

import { ActionSetting } from './ActionSetting';

export class GeneralSetting extends React.Component<{}> {
    render() {
        return (
            <>
                <Divider orientation='left'>General Information</Divider>
                <Divider />
                <ActionSetting />
            </>
        );
    }
}
