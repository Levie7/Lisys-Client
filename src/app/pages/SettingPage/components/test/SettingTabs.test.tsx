import { shallow } from 'enzyme';
import React from 'react';

import { TabPositions } from 'src/shared/components/Tabs';

import { SettingTabs } from '../SettingTabs';

describe('SettingTabs', () => {
    const props: { position: TabPositions } = { position: 'top' };
    const wrap = shallow(<SettingTabs {...props} />);

    it('<SettingTabs/> - should render SettingTabs component correctly', () => {
        expect(wrap).toMatchSnapshot();
    });

    it('should render setting tabs children component', () => {
        expect(wrap.find('Tabs').exists()).toBeTruthy();
        expect(wrap.find('TabPane').exists()).toBeTruthy();
        expect(wrap.find('Memo()').exists()).toBeTruthy();
    });
});
