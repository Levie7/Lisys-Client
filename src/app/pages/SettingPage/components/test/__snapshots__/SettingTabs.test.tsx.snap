// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingTabs <SettingTabs/> - should render SettingTabs component correctly 1`] = `ShallowWrapper {}`;
